modules/launcher/src/main/java/org/mule/runtime/module/launcher/log4j2/LoggerContextConfigurer.java

```diff
@@ -186,13 +186,19 @@ private void addDefaultArtifactContext(MuleLoggerContext context) {
       // configure a default one
       if (isUrlInsideDirectory(context.getConfigFile(), MuleContainerBootstrapUtils.getMuleConfDir())) {
         if (!context.isStopping() && !hasFileAppender(context)) {
-          addDefaultAppender(context, logFile.getAbsolutePath());
-          removeConsoleAppender(context);
+          addDefaultAppenderWithNoConsole(context, logFile);
         }
+      } else if (!context.isStopping() && context.getConfiguration().getAppenders().isEmpty()) {
```

Comment:
the condition `!context.isStopping()` is repeated, as well as the call to `addDefaultAppenderWithNoConsole(context, logFile);`
Refactor to simplify